### PR TITLE
Integrate first code quality analysis outcomes from SonarQube (Nemo)

### PR DESCRIPTION
**Pull request by acopet**
Resolve SonarQube issues detected by first Nemo run. Improve Code coverage.
Linked to issue #2.

### CODE REVIEW

File: src/test/java/org/sonar/plugins/stash/StashIssueReportingPostJobTest.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    when(config.hasToNotifyStash()).thenReturn(true);
+    
+    SonarQubeIssuesReport report = mock(SonarQubeIssuesReport.class);
+    when(report.countIssues()).thenReturn(10);
+    when(stashRequestFacade.extractIssueReport(projectIssues, inputFileCache)).thenReturn(report);
+    
+    int issueThreshold = 100;
+    when(stashRequestFacade.getIssueThreshold()).thenReturn(issueThreshold);
+    
+    StashIssueReportingPostJob job = new StashIssueReportingPostJob(config, projectIssues, inputFileCache, stashRequestFacade);
```
Comment (mathieusimon) [inline src/test/java/org/sonar/plugins/stash/StashIssueReportingPostJobTest.java]: Declare it as a member of the class. We will gain in readability.
To be generalized on others test classes.

File: src/test/java/org/sonar/plugins/stash/StashIssueReportingPostJobTest.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    job.executeOn(project, context);
+    
+    verify(stashRequestFacade, times(1)).postCommentPerIssue(eq(STASH_PROJECT), eq(STASH_REPOSITORY), eq(STASH_PULLREQUEST_ID), eq(SONARQUBE_URL), eq(report), (StashClient) Mockito.anyObject());
+    verify(stashRequestFacade, times(1)).postAnalysisOverview(eq(STASH_PROJECT), eq(STASH_REPOSITORY), eq(STASH_PULLREQUEST_ID), eq(SONARQUBE_URL), eq(issueThreshold), eq(report), (StashClient) Mockito.anyObject());
+  }
+  
+  @Test
+  public void testExecuteOnWithReachedTheshold() throws Exception {
```
Comment (mathieusimon) [inline src/test/java/org/sonar/plugins/stash/StashIssueReportingPostJobTest.java]: typo in method name

File: src/test/java/org/sonar/plugins/stash/StashRequestFacadeTest.java [hunk 3]
```diff
@@ -21,14 +35,79 @@
...
+  @Mock
+  StashDiffReport diffReport;
+  
+  @Mock
+  StashCommentReport stashComments1;
+  
+  @Mock
+  StashCommentReport stashComments2;
+  
   private StashRequestFacade facade;
```
Comment (mathieusimon) [inline src/test/java/org/sonar/plugins/stash/StashRequestFacadeTest.java]: Must be on top of the declaration as it is the instance of the class tested.

File: src/test/java/org/sonar/plugins/stash/issue/collector/SonarQubeCollectorTest.java [hunk 4]
```diff
@@ -142,4 +142,24 @@ public void testExtractIssueReportWithOldOption() {
     assertTrue(report.countIssues("severity1") == 0);
     assertTrue(report.countIssues("severity2") == 1);
   }
+  
+  @Test
+  public void testExtractIssueReportWithNoInputFile(){
```
Comment (mathieusimon) [inline src/test/java/org/sonar/plugins/stash/issue/collector/SonarQubeCollectorTest.java]: The test name is not relevant enough to reflect what it is really verified here.
We are checking that the process carry on even if an inputFile is null.

File: src/test/java/org/sonar/plugins/stash/StashIssueReportingPostJobTest.java (hunk 1 above)
Comment (acopet) [inline src/test/java/org/sonar/plugins/stash/StashIssueReportingPostJobTest.java]: Done!

File: src/test/java/org/sonar/plugins/stash/StashIssueReportingPostJobTest.java (hunk 2 above)
Comment (acopet) [inline src/test/java/org/sonar/plugins/stash/StashIssueReportingPostJobTest.java]: Done!

File: src/test/java/org/sonar/plugins/stash/StashRequestFacadeTest.java (hunk 3 above)
Comment (acopet) [inline src/test/java/org/sonar/plugins/stash/StashRequestFacadeTest.java]: Done!

File: src/test/java/org/sonar/plugins/stash/issue/collector/SonarQubeCollectorTest.java (hunk 4 above)
Comment (acopet) [inline src/test/java/org/sonar/plugins/stash/issue/collector/SonarQubeCollectorTest.java]: Done!

**Pull request merged**